File: API/Entities/CustomerOtherDetails.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities
{
    public class CustomerOtherDetails:BaseEntity
    {
        public Guid CustomerId { get; set; }
        public virtual Customer Customer { get; set; }
        public string Currency { get; set; }
        public Guid? TaxId { get; set; }
        public virtual Tax Tax { get; set; }
        public Guid? TermsId { get; set; }
        public virtual Terms Terms { get; set; }
        public EnablePortal EnablePortal { get; set; }
        public string Facebook { get; set; }
        public string Twitter { get; set; }
       
    }
    public enum EnablePortal
    {
        Yes,
        No,
    }
}

File: Client/src/app/_model/customerOtherDetails.ts
import { Guid } from 'guid-typescript';
import { Tax } from './tax';
import { Term } from './term';

export interface OtherDetails {
  id: Guid;
  customerId: Guid;
  currency: string;
  tax: Tax;
  terms: Term;
  enablePortal: EnablePortal;
  facebook: string;
  twitter: string;
}

export enum EnablePortal {
  Yes,
  No,
}
